### PR TITLE
Expose postfx to editor and scripts #696

### LINKED ISSUE #696: Expose postfx to editor and scripts
i want to be able to influence the postfx pipeline from the editor and scripts

### PR DESCRIPTION
**Pull request by LarsIngo**
Fixes #696

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L359](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L359): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L369](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L369): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L387](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L387): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L406](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L406): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L426](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L426): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L444): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L463](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L463): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L484](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/98e4f34a86295b2503f18d1b4a87e625d94a2c59/src/Engine/Manager/RenderManager.cpp#L484): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/RenderManager.hpp [hunk 1]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
...
+        ENGINE_API void SetGamma(float gamma);
+
+        /// Gamma correction.
+        /**
+         * @return gamma Gamma value.
+         */
+        ENGINE_API float GetGamma() const;
+
+
+        /// Set whether fog is applyed.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: applied*

File: src/Engine/Manager/RenderManager.hpp [hunk 2]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
...
+        ENGINE_API bool GetFogApply() const;
+
+
+        /// Set fog density.
+        /**
+         * @param fogDensity Density of the fog.
+         */
+        ENGINE_API void SetFogDensity(float fogDensity);
+
+        /// get fog density.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: Get*

File: src/Engine/Manager/RenderManager.hpp [hunk 3]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
         /// Remove all killed components.
         ENGINE_API void ClearKilledComponents();
         
+        /// Gamma correction.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: Should say that it's a setter

File: src/Engine/Manager/RenderManager.hpp [hunk 4]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
...
+        /**
+         * @param gamma Gamma to apply.
+         */
+        ENGINE_API void SetGamma(float gamma);
+
+        /// Gamma correction.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: Should say that it's a getter

File: src/Engine/Manager/RenderManager.hpp [hunk 5]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
...
+        ENGINE_API void SetFogColor(const glm::vec3& fogColor);
+
+        /// Get fog color.
+        /**
+         * @return Color of the fog.
+         */
+        ENGINE_API glm::vec3 GetFogColor() const;
+
+
+        /// Set whether color filter is applyed.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: applied*

File: src/Engine/Manager/RenderManager.hpp [hunk 6]
```diff
@@ -182,6 +182,96 @@ class RenderManager {
...
+        ENGINE_API void SetColorFilterColor(const glm::vec3& colorFilterColor);
+
+        /// Get color filer color.
+        /**
+         * @return Color of the color filter.
+         */
+        ENGINE_API glm::vec3 GetColorFilterColor() const;
+
+
+        /// Set whether dithering is applyed.
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.hpp]: applied*

File: src/Video/RenderProgram/StaticRenderProgram.hpp [hunk 7]
```diff
@@ -55,6 +55,109 @@ namespace Video {
...
+            void SetGamma(float gamma);
+
+            /// Gamma correction.
+            /**
+             * @return gamma Gamma value.
+             */
+            float GetGamma() const;
+
+
+            /// Set whether fog is applyed.
```
Comment (Chainsawkitten) [inline src/Video/RenderProgram/StaticRenderProgram.hpp]: applied*

File: src/Video/RenderProgram/StaticRenderProgram.hpp [hunk 8]
```diff
@@ -55,6 +55,109 @@ namespace Video {
...
+            void SetFogColor(const glm::vec3& fogColor);
+
+            /// Get fog color.
+            /**
+             * @return Color of the fog.
+             */
+            glm::vec3 GetFogColor() const;
+
+
+            /// Set whether color filter is applyed.
```
Comment (Chainsawkitten) [inline src/Video/RenderProgram/StaticRenderProgram.hpp]: applied*

File: src/Video/RenderProgram/StaticRenderProgram.hpp [hunk 9]
```diff
@@ -55,6 +55,109 @@ namespace Video {
...
+            void SetColorFilterColor(const glm::vec3& colorFilterColor);
+
+            /// Get color filer color.
+            /**
+             * @return Color of the color filter.
+             */
+            glm::vec3 GetColorFilterColor() const;
+
+
+            /// Set whether dithering is applyed.
```
Comment (Chainsawkitten) [inline src/Video/RenderProgram/StaticRenderProgram.hpp]: applied*

File: src/Video/Renderer.hpp [hunk 10]
```diff
@@ -134,6 +111,109 @@ namespace Video {
...
+            VIDEO_API void SetGamma(float gamma);
+
+            /// Gamma correction.
+            /**
+             * @return gamma Gamma value.
+             */
+            VIDEO_API float GetGamma() const;
+
+
+            /// Set whether fog is applyed.
```
Comment (Chainsawkitten) [inline src/Video/Renderer.hpp]: applied*

File: src/Video/Renderer.hpp [hunk 11]
```diff
@@ -134,6 +111,109 @@ namespace Video {
...
+            VIDEO_API void SetFogColor(const glm::vec3& fogColor);
+
+            /// Get fog color.
+            /**
+             * @return Color of the fog.
+             */
+            VIDEO_API glm::vec3 GetFogColor() const;
+
+
+            /// Set whether color filter is applyed.
```
Comment (Chainsawkitten) [inline src/Video/Renderer.hpp]: applied*

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L359](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L359): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L369](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L369): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L387](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L387): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L406](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L406): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L426](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L426): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L444): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L463](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L463): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L484](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2e480cbe16c2b8ba7ddf04a26d7d11b45f83de3/src/Engine/Manager/RenderManager.cpp#L484): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L359](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L359): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L369](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L369): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L387](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L387): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L406](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L406): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L426](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L426): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L444): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L463](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L463): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L484](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f627b738f5f3f40ba5249caa408cfd67dc9fc930/src/Engine/Manager/RenderManager.cpp#L484): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L359](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L359): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L369](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L369): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L387](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L387): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L406](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L406): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L426](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L426): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L444): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L463](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L463): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L484](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/de91821a1d35495345a786ca1194b437b5a9f871/src/Engine/Manager/RenderManager.cpp#L484): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L360](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L360): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L370](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L370): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L388](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L388): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L407](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L407): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L427](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L427): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L445](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L445): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L464](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L464): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L485](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/9cac2c5de6de52dd7fc0d2056d074be18c068acc/src/Engine/Manager/RenderManager.cpp#L485): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L183](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Hymn.cpp#L183): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L360](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L360): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L370](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L370): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L388](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L388): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L407](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L407): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L427](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L427): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L445](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L445): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L464](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L464): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L485](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/91c07400207ca176d348bbbce01318ca8eefd433/src/Engine/Manager/RenderManager.cpp#L485): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**